src/integrationTest/java/uk/gov/hmcts/darts/annotation/AnnotationControllerTest.java

```diff
@@ -0,0 +1,302 @@
+
+    private final BasicJsonTester json = new BasicJsonTester(getClass());
+
+    @Mock
+    private DataManagementApi dataManagementApi;
+
+    @Mock
+    private InputStream inputStreamResource;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private AnnotationTestGivensBuilder given;
+
+
+    @Test
+    void returnsAnnotationId() throws Exception {
+
+        createAuthenticatedJudgeWithGlobalAccessAndEmail("judge@global.com");
+
+        var mvcResult = mockMvc.perform(
+                multipart(ENDPOINT)
+                    .file(someAnnotationPostDocument())
+                    .file(someAnnotationPostBodyFor(createSomeMinimalHearing())))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        var response = mvcResult.getResponse().getContentAsString();
+        assertThat(json.from(response)).hasJsonPathNumberValue("annotation_id");
+    }
+
+    @Test
+    void allowsJudgeWithGlobalAccessToUploadAnnotations() throws Exception {
+
+        createAuthenticatedJudgeWithGlobalAccessAndEmail("judge@global.com");
+
+        mockMvc.perform(
+                multipart(ENDPOINT)
+                    .file(someAnnotationPostDocument())
+                    .file(someAnnotationPostBodyFor(createSomeMinimalHearing())))
+            .andExpect(status().isOk())
+            .andReturn();
+    }
+
+    @Test
+    void allowsJudgeAuthorisedForCourthouseAccessToUploadAnnotations() throws Exception {
+        var hearing = dartsDatabase.save(createSomeMinimalHearing());
+        createAuthenticatedJudgeAuthorizedForCourthouse("judge@global.com", hearing.getCourtroom().getCourthouse());
+
+        mockMvc.perform(
+                multipart(ENDPOINT)
+                    .file(someAnnotationPostDocument())
+                    .file(someAnnotationPostBodyFor(hearing)))
+            .andExpect(status().isOk())
+            .andReturn();
+    }
+
+    @Test
+    void shouldThrowHttp404ForValidJudgeAndInvalidAnnotationDocumentEntity() throws Exception {
+
+
+        someAnnotationCreatedBy(given.anAuthenticatedUserWithGlobalAccessAndRole(JUDGE));
+
+        var binaryData = mock(BinaryData.class);
+        lenient().when(dataManagementApi.getBlobDataFromInboundContainer(any(UUID.class))).thenReturn(binaryData);
+        lenient().when(binaryData.toStream()).thenReturn(inputStreamResource);
+
+        MockHttpServletRequestBuilder requestBuilder = get(ANNOTATION_DOCUMENT_ENDPOINT, 1, 1);
+
+        mockMvc.perform(
+                requestBuilder)
+            .andExpect(status().isNotFound())
+            .andExpect(jsonPath("$.type").value("ANNOTATION_104"))
+            .andExpect(jsonPath("$.title").value("invalid annotation id or annotation document id"))
+            .andExpect(jsonPath("$.status").value("404"))
+            .andReturn();
+    }
+
+    @Test
+    @Transactional
```

Comment:
We dont want to introduce more transactional tests.  What is the reason for it?  If not sure I can take a look.

Reply:
Hi Tony. Detached entity. So ran the test with a transaction. Also had to use dirties context to clear data between  test methods